crates/cust/src/memory/device/device_slice.rs

```diff
@@ -221,14 +236,15 @@ impl<T: DeviceCopy + Pod> DeviceSlice<T> {
     /// In total it will set `sizeof<T> * len` values of `value` contiguously.
     #[cfg_attr(docsrs, doc(cfg(feature = "bytemuck")))]
     pub fn set_8(&mut self, value: u8) -> CudaResult<()> {
-        if self.ptr.is_null() {
-            return Ok(());
-        }
-
```

Comment:
Why were these checks removed? They are needed for ZST device slices, since zero-width allocation in cuda is not possible (and makes literally no sense)

Reply:
These checks only checked if the pointer itself was null, which is not actually possible with references. I'll replace them with length checks then.

Reply:
Hmmm not sure if that will work because zst deviceslices still must track their length correctly

Reply:
Honestly i think we should just straight up remove support for ZST device allocations, they make no sense, there is no reason why you would want to pass a ZST to the GPU in the form of a device slice. In that case we can insert an assert on allocation that ensures the type is not a ZST, either that or return an InvalidAllocation error